doc/languages-frameworks/gradle.section.md

```diff
@@ -0,0 +1,137 @@
+# Gradle {#gradle}
+
+Gradle is a popular build tool for Java/Kotlin. Unfortunately, despites
+its ubiquity, it has many poor design decisions. Because Gradle itself
+doesn't offer reproduced builds in any meaningful way, nixpkgs has a
+man-in-the-middle-proxy designed for intercepting Gradle web requests.
+
+## Building a Gradle package {#building-a-gradle-package}
+
+A simple case is when the package doesn't have any dependencies. In that
+case, simply adding `gradle` to `nativeBuildInputs` may be enough:
+
+```nix
+stdenv.mkDerivation {
+  pname = "jextract";
+  version = "unstable-2023-11-27";
+
+  src = fetchFromGitHub {
+    owner = "openjdk";
+    repo = "jextract";
+    rev = "8730fcf05c229d035b0db52ee6bd82622e9d03e9";
+    hash = "sha256-Wct/yx5C0EjDtDyXNYDH5LRmrfq7islXbPVIGBR6x5Y=";
+  };
+
+  nativeBuildInputs = [
+    gradle
+    makeWrapper
+  ];
+
+  gradleFlags = [
+    "-Pllvm_home=${llvmPackages.libclang.lib}"
+    "-Pjdk21_home=${jdk21}"
+  ];
+
+  doCheck = true;
+
+  gradleCheckTask = "verify";
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/opt/
+    cp -r ./build/jextract $out/opt/jextract
+
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    makeWrapper "$out/opt/jextract/bin/jextract" "$out/bin/jextract"
```

Comment:
If people would ask if opt if prefered over share, what would we tell them?

Reply:
For the purpose of documentation, you could just make it an obvious placeholder like `$out/path/to/jextract`.

Reply:
i did just that